js/data/symbol_bucket.js

```diff
@@ -36,10 +36,16 @@ function SymbolBucket(options) {
     var zoomHistory = { lastIntegerZoom: Infinity, lastIntegerZoomTime: 0, lastZoom: 0 };
 
     this.adjustedTextMaxSize = this.layer.getLayoutValue('text-size', 18, zoomHistory);
-    this.adjustedTextSize = this.layer.getLayoutValue('text-size', this.zoom + 1, zoomHistory);
+    this.adjustedTextSizes = [
+        this.layer.getLayoutValue('text-size', this.zoom, zoomHistory),
+        this.layer.getLayoutValue('text-size', this.zoom + 1, zoomHistory)
+    ];
 
     this.adjustedIconMaxSize = this.layer.getLayoutValue('icon-size', 18, zoomHistory);
-    this.adjustedIconSize = this.layer.getLayoutValue('icon-size', this.zoom + 1, zoomHistory);
+    this.adjustedIconSizes = [
+        this.layer.getLayoutValue('icon-size', this.zoom, zoomHistory),
+        this.layer.getLayoutValue('icon-size', this.zoom + 1, zoomHistory)
+    ];
```

Comment:
Yes, if a value between z and z+1 is bigger or smaller than both of the two values at z and z + 1 then this solution isn't ideal. Having a `getMaxLayoutValue` and a `getMinLayoutValue` would be more correct but I'm not sure it would make a significant enough difference to be worth the extra complexity.

For this to make an actual difference in practice the difference between the maximum size between z and z + 1 would need to be significantly bigger than the maximum of the value at z and z + 1.

It might be worth it. What do you think? If it's worth it, do you want to take a shot at adding this?